sacc/data_types.py

```diff
@@ -0,0 +1,404 @@
+    "cmbCluster_temperatureDensity_cl": ['ell'],
+    "cmbCluster_temperatureDensity_xi": ['theta'],
+    "cmbGalaxy_convergenceDensity_cl": ['ell'],
+    "cmbGalaxy_convergenceDensity_xi": ['theta'],
+    "cmbGalaxy_convergenceShear_cl_b": ['ell'],
+    "cmbGalaxy_convergenceShear_cl_e": ['ell'],
+    "cmbGalaxy_convergenceShear_xi_minus": ['theta'],
+    "cmbGalaxy_convergenceShear_xi_plus": ['theta'],
+    "cmbGalaxy_convergence_cl": ['ell'],
+    "cmbGalaxy_convergence_xi": ['theta'],
+    "cmbGalaxy_polarizationConvergence_cl_b": ['ell'],
+    "cmbGalaxy_polarizationConvergence_cl_e": ['ell'],
+    "cmbGalaxy_polarizationDensity_cl_b": ['ell'],
+    "cmbGalaxy_polarizationDensity_cl_e": ['ell'],
+    "cmbGalaxy_polarizationDensity_xi_t": ['theta'],
+    "cmbGalaxy_polarizationDensity_xi_x": ['theta'],
+    "cmbGalaxy_polarizationShear_cl_bb": ['ell'],
+    "cmbGalaxy_polarizationShear_cl_be": ['ell'],
+    "cmbGalaxy_polarizationShear_cl_eb": ['ell'],
+    "cmbGalaxy_polarizationShear_cl_ee": ['ell'],
+    "cmbGalaxy_polarizationShear_xi_minus": ['theta'],
+    "cmbGalaxy_polarizationShear_xi_plus": ['theta'],
+    "cmbGalaxy_temperatureConvergence_cl": ['ell'],
+    "cmbGalaxy_temperatureConvergence_xi": ['theta'],
+    "cmbGalaxy_temperatureDensity_cl": ['ell'],
+    "cmbGalaxy_temperatureDensity_xi": ['theta'],
+    "cmbGalaxy_temperatureShear_cl_tb": ['ell'],
+    "cmbGalaxy_temperatureShear_cl_te": ['ell'],
+    "cmbGalaxy_temperatureShear_xi_tb": ['theta'],
+    "cmbGalaxy_temperatureShear_xi_te": ['theta'],
+    "cmb_convergence_cl": ['ell'],
+    "cmb_convergence_xi": ['theta'],
+    "cmb_polarization_cl_bb": ['ell'],
+    "cmb_polarization_cl_be": ['ell'],
+    "cmb_polarization_cl_eb": ['ell'],
+    "cmb_polarization_cl_ee": ['ell'],
+    "cmb_polarization_xi_minus": ['theta'],
+    "cmb_polarization_xi_plus": ['theta'],
+    "cmb_temperaturePolarization_cl_t": ['ell'],
+    "cmb_temperaturePolarization_xi_t": ['theta'],
+    "cmb_temperaturePolarization_xi_x": ['theta'],
+    "cmb_temperature_cl_tt": ['ell'],
+    "galaxy_convergenceDensity_cl": ['ell'],
+    "galaxy_convergenceDensity_xi": ['theta'],
+    "galaxy_convergence_cl": ['ell'],
+    "galaxy_convergence_xi": ['theta'],
+    "galaxy_density_cl": ['ell'],
+    "galaxy_density_xi": ['theta'],
+    "galaxy_shearConvergence_cl_b": ['ell'],
+    "galaxy_shearConvergence_cl_e": ['ell'],
+    "galaxy_shearConvergence_xi_t": ['theta'],
+    "galaxy_shearConvergence_xi_x": ['theta'],
+    "galaxy_shearDensity_cl_b": ['ell'],
+    "galaxy_shearDensity_cl_e": ['ell'],
+    "galaxy_shearDensity_xi_t": ['theta'],
+    "galaxy_shearDensity_xi_x": ['theta'],
+    "galaxy_shear_cl_bb": ['ell'],
+    "galaxy_shear_cl_be": ['ell'],
+    "galaxy_shear_cl_eb": ['ell'],
+    "galaxy_shear_cl_ee": ['ell'],
+    "galaxy_shear_cosebi_bb": [],
+    "galaxy_shear_cosebi_ee": [],
+    "galaxy_shear_xi_minus": ['theta'],
+    "galaxy_shear_xi_plus": ['theta'],
+}
```

Comment:
Thanks a lot for the documentation for additional types!

About the official ones, sorry, more nags!
I think you're missing
```
cmbCluster_convergenceDensity_<cl/xi>
cmb_polarizationConvergence_<cl/xi>_<e/b/t/x>
cmb_temperatureConvergence_<cl/xi>
cmbGalaxy_polarizationConvergence_xi_<t/x>
```

And questions:
- why clusterGalaxy_densityShear_cl_tb but cmbCluster_polarizationDensity_cl_b? I think there are others like this. I think you're aiming for spin0-spin2 correlation functions to be xi_t and xi_x.
- Similar: what is "cmbGalaxy_temperatureShear_xi_tb" and "cmbGalaxy_temperatureShear_xi_te"? I guess they should be "cmbGalaxy_temperatureShear_xi_t/x"?
- In principle, spin2-spin2 things like "cmbGalaxy_polarizationShear_xi_minus" have imaginary components corresponding to the parity-violating terms (C_ell^EB). E.g. see Eq. 13 in https://arxiv.org/pdf/astro-ph/0303414.pdf
- You have "cmb_temperaturePolarization_cl_t" only, but you should have "cmb_temperaturePolarization_cl_e" and "cmb_temperaturePolarization_cl_b".
- I think "cmb_temperature_cl_tt" should just be "cmb_temperature_cl"
- You only have cosebis for shear. Should they be there for CMB polarization too?

Reply:
I'm fine with keeping the imaginary parts out if you want to, BTW, just thought it was worth bringing it up, since the EB/BE power spectra are there.

Reply:
> I think you're missing

Thanks, added them all!

> why clusterGalaxy_densityShear_cl_tb but cmbCluster_polarizationDensity_cl_b?

Fixed in that case, and the others

> what is "cmbGalaxy_temperatureShear_xi_tb" and "cmbGalaxy_temperatureShear_xi_te"? 

Thanks - fixed (remove b/e)

> In principle, spin2-spin2 things like "cmbGalaxy_polarizationShear_xi_minus" have imaginary components

Added imagPlus and imagMinus variants - might be useful.

> You have "cmb_temperaturePolarization_cl_t" only, but you should have "cmb_temperaturePolarization_cl_e" and "cmb_temperaturePolarization_cl_b".

Changed.

> I think "cmb_temperature_cl_tt" should just be "cmb_temperature_cl"

Agreed - changed.

>  You only have cosebis for shear. Should they be there for CMB polarization too?

Added!

Reply:
Thanks @joezuntz . Two tiny ones:

You have `cmbGalaxy_temperatureShear_xi_t` twice. I think the second one should be `cmbGalaxy_temperatureShear_xi_x`

Also, if `polarization` and `shear` have cosebis, then `polarizationShear` should probably have them too (although I don't care enough about cosebis to push this one).

Reply:
Thanks - fix pushed for both

Reply:
Thanks!